bgpd/bgp_fsm.c

```diff
@@ -1863,6 +1863,34 @@ static int bgp_establish(struct peer *peer)
 			}
 		}
 
+	if (!CHECK_FLAG(peer->cap, PEER_CAP_RESTART_RCV)) {
+		if ((bgp_peer_gr_mode_get(peer) == PEER_GR)
+		    || ((bgp_peer_gr_mode_get(peer) == PEER_GLOBAL_INHERIT)
+			&& (bgp_global_gr_mode_get(peer->bgp) == GLOBAL_GR))) {
+			for (afi = AFI_IP; afi < AFI_MAX; afi++)
+				for (safi = SAFI_UNICAST; safi <= SAFI_MPLS_VPN;
+				     safi++)
```

Comment:
Can we use `FOREACH_AFI_SAFI` here?

Reply:
Thanks, @qlyoung. I've made the changes.